### PR TITLE
Calls

### PR DESCRIPTION
**Pull request by NicoBuchhalter**
Added Calls to the SDK and refactor a little bit.

### CODE REVIEW

File: Examples/Call/CreateCall.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            {
+                source = "31644556677",
+                destination = "33766723144",
+                callFlow = newCallFlow
+
+            };
+            try
+            {
+                var callResponse = client.CreateCall(newCall);
+                var call = callResponse.Data.FirstOrDefault();
```
Comment (jen-obyrne) [inline Examples/Call/CreateCall.cs]: Think you need to add a using for `System.Linq` or this project may not build?

File: Examples/Call/CreateCall.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
+using System;
+using System.IO;
+using MessageBird;
+using MessageBird.Exceptions;
+using System.Collections.Generic;
+using MessageBird.Objects.Voice;
```
Comment (jen-obyrne) [inline Examples/Call/CreateCall.cs]: Very minor, but can you sort your usings?

File: Examples/Call/CreateCall.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+namespace Examples.Call
+{
+    class CreateCall
+    {
+        const string YourAccessKey = "YOUR_ACCESS_KEY"; // your access key here.
+
+        static void Main(string[] args)
```
Comment (jen-obyrne) [inline Examples/Call/CreateCall.cs]: set the accessibility level on this method

File: MessageBird/Client.cs [hunk 4]
```diff
@@ -155,15 +155,80 @@ public void DeleteCallFlow(string id)
...
+        /// <param name="limit">Set how many records will return from the server</param>
+        /// <param name="offset">Identify the starting point to return rows from a result</param>
+        /// <returns>If successful, this request will return an object with a data, _links and pagination properties.</returns>
+        public CallList ListCalls(int limit = 20, int offset = 0)
+        {
+            var resource = new CallLists();
+
+            var list = (CallList)resource.Object;
+            list.Limit = limit;
+            list.Offset = offset;
```
Comment (jen-obyrne) [inline MessageBird/Client.cs]: Do you need these lines? Looks like `list` isn't being used anywhere

File: MessageBird/Client.cs [hunk 5]
```diff
@@ -155,15 +155,80 @@ public void DeleteCallFlow(string id)
...
+
+            restClient.Retrieve(resource);
+
+            return (CallList)resource.Object;
```
Comment (jen-obyrne) [inline MessageBird/Client.cs]: Is this method working how you expect? Looks like the Retrieve method returns something that you don't use

File: MessageBird/Client.cs [hunk 6]
```diff
@@ -155,15 +155,80 @@ public void DeleteCallFlow(string id)
...
+            var resource = new Calls(new Call { Id = callId });
+            restClient.Retrieve(resource);
+
+            return (VoiceResponse<Call>)resource.Object;
+        }
+
+        /// <summary>
+        /// This request deletes a call. The parameters are the unique ID of the call, the leg and the call with which the call is associated.
+        /// If successful, this request will return an HTTP header of 204 No Content and an empty response.
+        /// If the request failed, an error object will be returned.
```
Comment (jen-obyrne) [inline MessageBird/Client.cs]: Can you update this comment? This method doesn't return an error that I can see

File: MessageBird/Objects/Voice/Webhook.cs [hunk 7]
```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
```
Comment (jen-obyrne) [inline MessageBird/Objects/Voice/Webhook.cs]: Can you please delete the unused using statements

File: MessageBird/Objects/Voice/Call.cs [hunk 8]
```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
```
Comment (jen-obyrne) [inline MessageBird/Objects/Voice/Call.cs]: Can you please delete the unused using statements here?

File: Tests/UnitTests/MessageBirdUnitTests/Resources/CallTest.cs [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+using System.Linq;
+using MessageBird.Objects.Voice;
+using MessageBird.Resources.Voice;
+using MessageBird.Exceptions;
+
+
+namespace MessageBirdUnitTests.Resources
+{
+    [TestClass]
+
```
Comment (jen-obyrne) [inline Tests/UnitTests/MessageBirdUnitTests/Resources/CallTest.cs]: Can you please delete this blank line?

File: Tests/UnitTests/MessageBirdUnitTests/Resources/CallTest.cs [hunk 10]
```diff
@@ -0,0 +1,135 @@
+using MessageBird;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections.Generic;
+using System;
+using System.IO;
+using System.Linq;
+using MessageBird.Objects.Voice;
+using MessageBird.Resources.Voice;
+using MessageBird.Exceptions;
```
Comment (jen-obyrne) [inline Tests/UnitTests/MessageBirdUnitTests/Resources/CallTest.cs]: Can you please remove the unused using statements?

File: MessageBird/Objects/Voice/Call.cs (hunk 8 above)
Comment (NicoBuchhalter) [reply MessageBird/Objects/Voice/Call.cs]: They are all being used actually

**Comment by jen-obyrne** [comment]
Thanks for making the changes. PR looks good 👍

**Pull request merged**